akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ClusterShardingImpl.scala

```diff
@@ -72,11 +73,36 @@ import akka.util.Timeout
   override def toString: String = delegate.toString
 }
 
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object EntityRefImpl {
+  /**
+   * Used for concatenating EntityTypeKey with entityId to construct unique persistenceId.
+   * This must be same as in Lagom, for compatibility.
```

Comment:
Unfortunately, the [Java](https://github.com/lagom/lagom/blob/cb2a7bb2cacbf84280fcc370c2505207146fe72a/persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/PersistentEntityActor.scala#L65) and [Scala](https://github.com/lagom/lagom/blob/cb2a7bb2cacbf84280fcc370c2505207146fe72a/persistence/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/PersistentEntityActor.scala#L89) Lagom APIs implement this differently. I'd hate to replicate the inconsistency, but I think we'll need to have some answer for compatibility. Maybe make the separator overridable per `EntityTypeKey`?

Reply:
ouch, thanks for noticing